server/src/main/java/org/elasticsearch/common/Numbers.java

```diff
@@ -126,9 +126,9 @@ public static long toLongExact(Number n) {
     /** Return the long that {@code stringValue} stores or throws an exception if the
      *  stored value cannot be converted to a long that stores the exact same
      *  value and {@code coerce} is false. */
-    public static long toLong(String stringValue, boolean coerce) {
+    public static Long toLong(String stringValue, boolean coerce) {
         try {
-            return Long.parseLong(stringValue);
+            return Long.valueOf(stringValue);
```

Comment:
I think I'd keep the return type on this one.

Reply:
I moved this to return the boxed type so that we do not rely on the JIT to inline+elide this boxing. Otherwise we would parse into a primitive `long`, then need to box when actually inserting into the set. I don't feel strongly about this, though.

Reply:
I thought about it more, and I changed back to primitive.